src/index.ts

```diff
@@ -0,0 +1,221 @@
+import * as logging from 'plylog';
+
+const logger = logging.getLogger('polymer-project-config');
+
+/**
+ * The default globs for matching all user application source files.
+ */
+export const defaultSourceGlobs = ['src/**/*'];
+
+/**
+ * Resolve any glob to the given path, even if glob
+ * is negative (begins with '!').
+ */
+function resolveGlob(fromPath: string, glob: string): string {
+  if (glob.startsWith('!')) {
+    const includeGlob = glob.substring(1);
+    return '!' + path.resolve(fromPath, includeGlob);
+  } else {
+    return path.resolve(fromPath, glob);
+  }
+}
+
+/**
+ * Given a user-provided options object, check for deprecated options. When one
+ * is found, warn the user and fix if possible.
+ */
+function fixDeprecatedOptions(options: any): ProjectOptions {
+  if (typeof options.sourceGlobs !== 'undefined') {
+    logger.warn('"sourceGlobs" config option has been renamed to "sources" and will no longer be supported in future versions');
+    options.sources = options.sources || options.sourceGlobs;
+  }
+  if (typeof options.includeDependencies !== 'undefined') {
+    logger.warn('"includeDependencies" config option has been renamed to "dependencies" and will no longer be supported in future versions');
+    options.dependencies = options.dependencies || options.includeDependencies;
+  }
+  return options;
+}
+
+
+export interface ProjectOptions {
+  /**
+   * Path to the root of the project on the filesystem. This can be an absolute
+   * path, or a path relative to the current working directory. Defaults to the
+   * current working directory of the process.
+   */
+  root?: string;
+
+  /**
+   * The path relative to `root` of the entrypoint file that will be served for
+   * app-shell style projects. Usually this is index.html.
+   */
+  entrypoint?: string;
+
+  /**
+   * The path relative to `root` of the app shell element.
+   */
+  shell?: string;
+
+  /**
+   * The path relative to `root` of the lazily loaded fragments. Usually the
+   * pages of an app or other bundles of on-demand resources.
+   */
+  fragments?: string[];
+
+  /**
+   * List of glob patterns, relative to root, of this project's sources to read
+   * from the file system.
+   */
+  sources?: string[];
+
+  /**
+   * List of file paths, relative to the project directory, that should be included
+   * as dependencies in the build target.
+   */
+  dependencies?: string[];
+}
+
+export class ProjectConfig {
+
+  readonly root: string;
+  readonly entrypoint: string;
+  readonly shell: string;
+  readonly fragments: string[];
+  readonly sources: string[];
+  readonly dependencies: string[];
+
+  readonly allFragments: string[];
+
+  /**
+   * Given an absolute file path to a polymer.json-like ProjectOptions object,
+   * read that file. If no file exists, null is returned. If the file exists
+   * but there is a problem reading or parsing it, throw an exception.
+   */
+  static loadOptionsFromFile(filepath: string): ProjectOptions {
```

Comment:
I see. It might still be nice to only expose loading of the config and let the config merge in new options. That means you could smartly merge lists is that what you wanted. No need to change now though.